v0/destinations/ga4/transform.js

```diff
@@ -0,0 +1,414 @@
+const get = require("get-value");
+const logger = require("../../../logger");
+const { EventType } = require("../../../constants");
+const {
+  CustomError,
+  defaultPostRequestConfig,
+  constructPayload,
+  defaultRequestConfig,
+  extractCustomFields,
+  isEmptyObject,
+  flattenJson,
+  getDestinationExternalID
+} = require("../../util");
+const {
+  ENDPOINT,
+  trackCommonConfig,
+  eventNameMapping,
+  mappingConfig,
+  ConfigCategory
+} = require("./config");
+const {
+  msUnixTimestamp,
+  isReservedEventName,
+  GA4_RESERVED_PARAMETER_EXCLUSION,
+  removeReservedParameterPrefixNames,
+  GA4_RESERVED_USER_PROPERTY_EXCLUSION,
+  removeReservedUserPropertyPrefixNames,
+  isReservedWebCustomEventName,
+  isReservedWebCustomPrefixName,
+  getDestinationItemProperties
+} = require("./utils");
+
+const trackResponseBuilder = async (message, { Config }) => {
+  let event = get(message, "event");
+  if (!event) {
+    throw new CustomError("Event name is required", 400);
+  }
+
+  // trim and replace spaces with "_"
+  event = event.trim().replace(/\s+/g, "_");
+
```

Comment:
Apart from calling lowerCase every time we can make the event lowercase here. WDYT?

Reply:
Due to custom events being case sensitive